### PR TITLE
Timestepping contribution guide

### PR DESCRIPTION
**Pull request by thomasgibson**
# Description

This PR adds a markdown script (similar to `IterativeMethods.md`) which details what needs to be considered when adding a new time-integrator (ODESolver) to the codebase.

Please feel free to comment and add suggestions if anyone thinks something is missing or needs further elaboration.

I have

- [x] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [x] Followed all necessary [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/format.jl`
- [x] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by charleskawczynski** [comment]
Looking great! Please also make sure to update the [corresponding dictionary entry](https://github.com/climate-machine/CLIMA/blob/master/docs/list_of_extending_clima.jl#L33).

**Comment by thomasgibson** [comment]
> Looking great! Please also make sure to update the [corresponding dictionary entry](https://github.com/climate-machine/CLIMA/blob/master/docs/list_of_extending_clima.jl#L33).

Oops! I'll fix that.

File: docs/src/ExtendingCLIMA/Numerics/ODESolvers/Timestepping.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+## Basic Template for an IMEX/Additive Runge-Kutta Method
+
+IMEX, or IMplicit-EXplicit, Runge-Kutta methods require a bit more
+attention. IMEX methods are typically constructed from
+Additively-partitioned Runge-Kutta (ARK) methods. For IMEX methods,
+the standard way is to consider an ARK method with **two** partitions:
+one explicit part, and one implicit part. The implicit part will require
+a linear solver.
+
+An ARK method with an explicit and implicit component will require **two***
```
Comment (charleskawczynski) [inline docs/src/ExtendingCLIMA/Numerics/ODESolvers/Timestepping.md:74]:
```suggestion
An ARK method with an explicit and implicit component will require **two**
```

Comment (thomasgibson) [reply docs/src/ExtendingCLIMA/Numerics/ODESolvers/Timestepping.md]: Done!

**Pull request merged**